examples/plot_gait.py

```diff
@@ -0,0 +1,241 @@
+
+###############################################################################
+# Note the use of ``reshape`` to shape the signal as per ``dicodile``
+# requirements: the shape of the signal should be
+# ``(n_channels, *sig_support)``.
+# Here, we have a single-channel time series.
+
+###############################################################################
+# Then, we run DiCoDiLe!
+
+D_hat, z_hat, pobj, times = dicodile(X, D_init, n_iter=3,
+                                     n_workers=4,
+                                     dicod_kwargs={"max_iter": 10000},
+                                     verbose=6,
+                                     window=True)
+
+
+print("[DiCoDiLe] final cost : {}".format(pobj))
+
+###############################################################################
+# We'll now display the initial and final dictionary side by side
+
+# normalize dictionaries
+normalized_D_init = D_init / D_init.max()
+normalized_D_hat = D_hat / D_hat.max()
+
+fig = display_dictionaries(normalized_D_init, normalized_D_hat)
+
+###############################################################################
+# We can order the dictionary patches by decreasing sum of the activations'
+# absolute values in the activations ``z_hat``, which, intuitively, gives
+# a measure of how they contribute to the reconstruction.
+
+sum_abs_val = np.sum(np.abs(z_hat), axis=-1)
+sum_abs_val
+
+
+""
+# we negate sum_abs_val to sort in decreasing order
+patch_indices = np.argsort(-sum_abs_val)
+
+fig_reordered = display_dictionaries(normalized_D_init[patch_indices],
+                                     normalized_D_hat[patch_indices])
+
+###############################################################################
+# Signal reconstruction
+# ^^^^^^^^^^^^^^^^^^^^^
+#
+# Now, let's reconstruct the original signal
+
+X_hat = reconstruct(z_hat, D_hat)
+
+""
+fig_hat, ax_hat = plt.subplots()
+ax_hat.plot(X[0][5000:5800],
+            label='right foot vertical acceleration (ORIGINAL)')
+ax_hat.plot(X_hat[0][5000:5800],
+            label='right foot vertical acceleration (RECONSTRUCTED)')
+ax_hat.set_xlabel('time (x10ms)')
+ax_hat.set_ylabel('acceleration ($m.s^{-2}$)')
+ax_hat.legend()
+
+###############################################################################
+# Check that our representation is indeed sparse:
+
+np.count_nonzero(z_hat)
+###############################################################################
+# Besides our visual check, a measure of how closely we're reconstructing the
+# original signal is the (normalized) cross-correlation. Let's compute this:
+
+np.correlate(X[0], X_hat[0]) / (
+    np.sqrt(np.correlate(X[0], X[0]) * np.correlate(X_hat[0], X_hat[0])))
+
+###############################################################################
+# Detecting steps
+# ---------------
+
+###############################################################################
+# Multichannel signals
+# --------------------
+#
```

Comment:
it's weird to have 2 sections at the same level without anything in between

Reply:
indeed, but that's because the content for the section "detecting steps" is yet missing